Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx

```diff
@@ -72,15 +72,152 @@ int ds2manu(int i)
   return refDs2manu_st345[i];
 }
 
+//=============
+// Classes for custom mapping implementation
+#define MCH_MAX_FEEID 64
+#define MCH_MAX_CRU_LINK 12
+#define LINKID_MAX 0x7FF
+
+// CRU mapping
+class MapCRU
+{
+  bool mInitialized = {false};
+  uint16_t mSolarMap[MCH_MAX_FEEID][MCH_MAX_CRU_LINK];
+
+ public:
+  MapCRU() = default;
+  ~MapCRU() = default;
+  bool load(std::string mapFile);
+  bool initialized() const { return mInitialized; }
+  std::optional<uint16_t> operator()(const FeeLinkId& feeLinkId) const;
+};
+
+bool MapCRU::load(std::string mapFile)
+{
+  std::ifstream file;
+  file.open(mapFile);
+  if (!file) {
+    std::cout << "[MapCRU::readMapping] can't open file " << mapFile << std::endl;
+    return false;
+  }
+
```

Comment:
Here one should initialize all the `mSolarMap` elements to `-1` before loading the custom mapping, so that (FEEID, LINK) pairs that are not specified in the mapping can be properly detected.